src/main/java/com/example/finalproject/controllers/PostController.java

```diff
@@ -0,0 +1,35 @@
+package com.example.finalproject.controllers;
+
+import com.example.finalproject.models.Post;
+import com.example.finalproject.service.PostService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.sql.Date;
+import java.time.LocalDate;
+import java.util.List;
+
+@RestController
+public class PostController {
+
+    private final PostService postService;
+
+    @Autowired
+    public PostController(PostService postService) {
+        this.postService = postService;
+    }
+
+    @GetMapping("/get")
+    public ResponseEntity<List<Post>> getPostsByParam(
+            @RequestParam(required = false) String author,
+            @RequestParam(required = false) String tag,
+            @RequestParam(name = "start_date", required = false)@DateTimeFormat(pattern = "dd.MM.yyyy") LocalDate startDate,
+            @RequestParam(name = "end_date", required = false)@DateTimeFormat(pattern = "dd.MM.yyyy") LocalDate endDate
+            ){
+        return ResponseEntity.status(201).body(postService.getPostsByParam(author, tag, startDate, endDate));
```

Comment:
status 200 is more appropriate here. 200 - OK, 201 - CREATED. 201 is appropriate for cases when we save something